src/renderers/shared/fiber/ReactFiber.js

```diff
@@ -398,10 +370,42 @@ function createFiberFromElementType(
       info,
     );
   }
+
+  if (__DEV__) {
+    fiber._debugSource = element._source;
+    fiber._debugOwner = element._owner;
+  }
+
+  fiber.expirationTime = expirationTime;
+
+  return fiber;
+};
+
+function createFiberFromFragment(
+  elements: ReactFragment,
+  internalContextTag: TypeOfInternalContext,
+  expirationTime: ExpirationTime,
+  key: null | string,
+): Fiber {
+  const fiber = createFiber(Fragment, key, internalContextTag);
+  fiber.type = REACT_FRAGMENT_TYPE;
```

Comment:
If you always compare against `.tag` instead of `.type` you can get rid of this assignment. We shouldn't unnecessarily use fields we don't really need and for the fragment tag this field is unnecessary.

Reply:
Would it be preferable to have a branch on `current.tag === Fragment && element.type === REACT_FRAGMENT_TYPE` at places where it would usually be a `current.type === element.type`? That the initial way I implemented this but @acdlite suggested I can add this field to not have to special case.

Reply:
yea.